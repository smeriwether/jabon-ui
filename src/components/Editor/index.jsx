import React from 'react';
import PropTypes from 'prop-types';
import AceEditor from 'react-ace';
import 'brace/mode/xml';
import 'brace/theme/tomorrow';
import 'brace/ext/language_tools';

const propTypes = {
  value: PropTypes.string,
  name: PropTypes.string.isRequired,
  readOnly: PropTypes.bool,
  onChange: PropTypes.func
};

const defaultProps = {
  readOnly: false,
  onChange: function() {}
};

const editorOptions = {
  enableBasicAutocompletion: true,
  enableLiveAutocompletion: true,
  showLineNumbers: true,
  tabSize: 2
};

const Editor = ({ value, name, readOnly, onChange }) => {
  return (
    <AceEditor
      mode="xml"
      theme="tomorrow"
      name={ name }
      height="100%"
      width="100%"
      value={ value }
      readOnly={ readOnly }
      showPrintMargin={ false }
      showGutter={ true }
      highlightActiveLine={ true }
      setOptions={ editorOptions }
      onChange={ onChange }
    />
  );
};

Editor.propTypes = propTypes;
Editor.defaultProps = defaultProps;

export default Editor;
