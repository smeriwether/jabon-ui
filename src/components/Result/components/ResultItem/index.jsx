import React from 'react';
import PropTypes from 'prop-types';

import Editor from '../../../Editor';
import { formatMilliseconds } from '../../../../utils/time';


const propTypes = {
  result: PropTypes.shape({
    id: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number
    ]).isRequired,
    succeeded: PropTypes.bool.isRequired,
    runtime: PropTypes.number.isRequired,
    content: PropTypes.string
  }).isRequired
};

const ResultItem = ({ result }) => {
  const status = result.succeeded ? 'Success' : 'Failed';
  const runtime = formatMilliseconds(result.runtime);
  const dataRole = `result-item-${result.id}`

  return (
    <div className='h-100 w-100 pa3 bg-white ba b--moon-gray' data-role={ dataRole }>
      <div className='h-10 w-100 pb2 flex justify-between items-center'>
        <div className={ result.succeeded ? 'b dark-green' : 'b dark-red' } data-role='result-status'>
          { status }
        </div>
        <div className='silver'>
          { runtime }
        </div>
      </div>

      <div className='h-90 w-100 ba b--moon-gray' data-role='result-editor'>
        <Editor name={ 'code-editor-' + result.id } readOnly={ true } value={ result.content } />
      </div>
    </div>
  );
};

ResultItem.propTypes = propTypes;

export default ResultItem;
