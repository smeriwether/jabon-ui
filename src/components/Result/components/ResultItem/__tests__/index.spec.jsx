import ResultItem from '../../ResultItem';

import Editor from '../../../../Editor';

describe('<ResultItem />', () => {
  const setup = (overrides) => {
    const props = {
      result: {
        id: 1,
        succeeded: true,
        runtime: 123
      },
      ...overrides
    };

    const wrapper = shallow(<ResultItem { ...props } />);

    return { wrapper };
  };

  it('should render the Editor with a readOnly prop', () => {
    const { wrapper } = setup();
    const editor = wrapper.find(Editor);
    expect(editor.prop('readOnly')).toBe(true);
  });

  context('when the result succeeds', () => {
    it('should match the snapshot', () => {
      const result = { id: 1, succeeded: true, runtime: 123 };
      const { wrapper } = setup({ result });
      expect(wrapper).toMatchSnapshot();
    });
  });

  context('when the result fails', () => {
    it('should match the snapshot', () => {
      const result = { id: 2, succeeded: false, runtime: 123456 };
      const { wrapper } = setup({ result });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
