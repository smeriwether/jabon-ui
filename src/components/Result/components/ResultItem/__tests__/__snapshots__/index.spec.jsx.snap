// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> when the result fails should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultItem
    result={
      Object {
        "id": 2,
        "runtime": 123456,
        "succeeded": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="h-10 w-100 pb2 flex justify-between items-center"
        >
          <div
            className="b dark-red"
            data-role="result-status"
          >
            Failed
          </div>
          <div
            className="silver"
          >
            2 minutes 3.46 seconds
          </div>
        </div>,
        <div
          className="h-90 w-100 ba b--moon-gray"
          data-role="result-editor"
        >
          <Editor
            name="code-editor-2"
            onChange={[Function]}
            readOnly={true}
            value={undefined}
          />
        </div>,
      ],
      "className": "h-100 w-100 pa3 bg-white ba b--moon-gray",
      "data-role": "result-item-2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="b dark-red"
              data-role="result-status"
            >
              Failed
            </div>,
            <div
              className="silver"
            >
              2 minutes 3.46 seconds
            </div>,
          ],
          "className": "h-10 w-100 pb2 flex justify-between items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Failed",
              "className": "b dark-red",
              "data-role": "result-status",
            },
            "ref": null,
            "rendered": "Failed",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2 minutes 3.46 seconds",
              "className": "silver",
            },
            "ref": null,
            "rendered": "2 minutes 3.46 seconds",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Editor
            name="code-editor-2"
            onChange={[Function]}
            readOnly={true}
            value={undefined}
          />,
          "className": "h-90 w-100 ba b--moon-gray",
          "data-role": "result-editor",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "code-editor-2",
            "onChange": [Function],
            "readOnly": true,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="h-10 w-100 pb2 flex justify-between items-center"
          >
            <div
              className="b dark-red"
              data-role="result-status"
            >
              Failed
            </div>
            <div
              className="silver"
            >
              2 minutes 3.46 seconds
            </div>
          </div>,
          <div
            className="h-90 w-100 ba b--moon-gray"
            data-role="result-editor"
          >
            <Editor
              name="code-editor-2"
              onChange={[Function]}
              readOnly={true}
              value={undefined}
            />
          </div>,
        ],
        "className": "h-100 w-100 pa3 bg-white ba b--moon-gray",
        "data-role": "result-item-2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="b dark-red"
                data-role="result-status"
              >
                Failed
              </div>,
              <div
                className="silver"
              >
                2 minutes 3.46 seconds
              </div>,
            ],
            "className": "h-10 w-100 pb2 flex justify-between items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Failed",
                "className": "b dark-red",
                "data-role": "result-status",
              },
              "ref": null,
              "rendered": "Failed",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2 minutes 3.46 seconds",
                "className": "silver",
              },
              "ref": null,
              "rendered": "2 minutes 3.46 seconds",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Editor
              name="code-editor-2"
              onChange={[Function]}
              readOnly={true}
              value={undefined}
            />,
            "className": "h-90 w-100 ba b--moon-gray",
            "data-role": "result-editor",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "code-editor-2",
              "onChange": [Function],
              "readOnly": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ResultItem /> when the result succeeds should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ResultItem
    result={
      Object {
        "id": 1,
        "runtime": 123,
        "succeeded": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="h-10 w-100 pb2 flex justify-between items-center"
        >
          <div
            className="b dark-green"
            data-role="result-status"
          >
            Success
          </div>
          <div
            className="silver"
          >
            123 milliseconds
          </div>
        </div>,
        <div
          className="h-90 w-100 ba b--moon-gray"
          data-role="result-editor"
        >
          <Editor
            name="code-editor-1"
            onChange={[Function]}
            readOnly={true}
            value={undefined}
          />
        </div>,
      ],
      "className": "h-100 w-100 pa3 bg-white ba b--moon-gray",
      "data-role": "result-item-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="b dark-green"
              data-role="result-status"
            >
              Success
            </div>,
            <div
              className="silver"
            >
              123 milliseconds
            </div>,
          ],
          "className": "h-10 w-100 pb2 flex justify-between items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Success",
              "className": "b dark-green",
              "data-role": "result-status",
            },
            "ref": null,
            "rendered": "Success",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "123 milliseconds",
              "className": "silver",
            },
            "ref": null,
            "rendered": "123 milliseconds",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Editor
            name="code-editor-1"
            onChange={[Function]}
            readOnly={true}
            value={undefined}
          />,
          "className": "h-90 w-100 ba b--moon-gray",
          "data-role": "result-editor",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "code-editor-1",
            "onChange": [Function],
            "readOnly": true,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="h-10 w-100 pb2 flex justify-between items-center"
          >
            <div
              className="b dark-green"
              data-role="result-status"
            >
              Success
            </div>
            <div
              className="silver"
            >
              123 milliseconds
            </div>
          </div>,
          <div
            className="h-90 w-100 ba b--moon-gray"
            data-role="result-editor"
          >
            <Editor
              name="code-editor-1"
              onChange={[Function]}
              readOnly={true}
              value={undefined}
            />
          </div>,
        ],
        "className": "h-100 w-100 pa3 bg-white ba b--moon-gray",
        "data-role": "result-item-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="b dark-green"
                data-role="result-status"
              >
                Success
              </div>,
              <div
                className="silver"
              >
                123 milliseconds
              </div>,
            ],
            "className": "h-10 w-100 pb2 flex justify-between items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Success",
                "className": "b dark-green",
                "data-role": "result-status",
              },
              "ref": null,
              "rendered": "Success",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "123 milliseconds",
                "className": "silver",
              },
              "ref": null,
              "rendered": "123 milliseconds",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Editor
              name="code-editor-1"
              onChange={[Function]}
              readOnly={true}
              value={undefined}
            />,
            "className": "h-90 w-100 ba b--moon-gray",
            "data-role": "result-editor",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "code-editor-1",
              "onChange": [Function],
              "readOnly": true,
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
